src/WebJobs.Extensions.DurableTask/Storage/StorageServiceClientProvider.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Azure.Core;
+using DurableTask.AzureStorage;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask.Options;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Storage
+{
+    internal abstract class StorageServiceClientProvider<TClient, TClientOptions, TConnectionOptions> : IStorageServiceClientProvider<TClient, TClientOptions>
+        where TClientOptions : ClientOptions
+        where TConnectionOptions : StorageServiceConnectionOptions
+    {
+        private readonly IConfigurationSection connectionSection;
+        private readonly AzureComponentFactory componentFactory;
+        private readonly AzureEventSourceLogForwarder logForwarder;
+
+        protected StorageServiceClientProvider(IConfigurationSection connectionSection, AzureComponentFactory componentFactory, AzureEventSourceLogForwarder logForwarder)
+        {
+            this.connectionSection = connectionSection ?? throw new ArgumentNullException(nameof(connectionSection));
+            this.componentFactory = componentFactory ?? throw new ArgumentNullException(nameof(componentFactory));
+            this.logForwarder = logForwarder ?? throw new ArgumentNullException(nameof(logForwarder));
+
+            this.logForwarder.Start();
+        }
+
+        public TClient CreateClient(TClientOptions options)
+        {
+            // Azure Functions allows for additional connection options on top of what is provided by the AzureComponentFactory
```

Comment:
I think I need to readjust this flow for V2+ functions that use connection strings to ensure the log forwarder starts properly

Reply:
Nevertheless, the flow should be:
1. Check for connection string
2. Check for AccountName/URI
3. Otherwise, rely on component factory to figure it out